### PR TITLE
EXPLAIN indexes=1 will print all KeyConditions instead of only the last one

### LINKED ISSUE #48237: Fix EXPLAIN indexes =1 for tuple IN (set)
Example:
```
create table tab (x UInt32, y UInt32) engine = MergeTree order by (x, y) settings index_granularity = 1;
insert into tab select number, number / 3 from numbers(6);

SELECT * FROM tab

┌─x─┬─y─┐
│ 0 │ 0 │
│ 1 │ 0 │
│ 2 │ 0 │
│ 3 │ 1 │
│ 4 │ 1 │
│ 5 │ 1 │
└───┴───┘


EXPLAIN indexes = 1
SELECT *
FROM tab
WHERE (x, y) IN ((0, 0), (5, 1))
FORMAT TSVRaw

Query id: e1b29671-7744-4818-8c94-7813dec17437

Expression ((Projection + Before ORDER BY))
  Filter (WHERE)
    ReadFromMergeTree (default.tab)
    Indexes:
      PrimaryKey
        Keys: 
          y
        Condition: (y in 2-element set)
        Parts: 1/1
        Granules: 3/6
```

**What's right**
PK condition is used for both `x` and `y` columns. `Granules: 3/6`

**What's wrong**
`EXPLAIN indexes = 1` prints only the last name from the tuple.

**Expected result**
```
Expression ((Projection + Before ORDER BY))
  Filter (WHERE)
    ReadFromMergeTree (default.tab)
    Indexes:
      PrimaryKey
        Keys: 
          x, y
        Condition: ((x, y) in 2-element set)
        Parts: 1/1
        Granules: 3/6
```

The reason: we remember only one column. Can be replaced to vector.
https://github.com/ClickHouse/ClickHouse/blob/66f546ef705f08ab895b310ac0be3483a2334f0a/src/Storages/MergeTree/KeyCondition.cpp#L1162

### PR DESCRIPTION
**Pull request by mgill25**
Changes KeyCondition's `out_key_column_num` to a vector, so that we can collect all column indexes and display them during `EXPLAIN indexes = 1`

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

- `EXPLAIN indexes=1` will now print all primary key conditions instead of just the last one. Fixes https://github.com/ClickHouse/ClickHouse/issues/48237

### Documentation entry for user-facing changes

```sql
EXPLAIN indexes = 1
SELECT *
FROM tab
WHERE (x, y) IN ((0, 0), (5, 1))
FORMAT TSVRaw

Query id: 3d5ea79f-07fe-45b4-833f-fdf80606e706

Expression ((Projection + Before ORDER BY))
  Filter (WHERE)
    ReadFromMergeTree (default.tab)
    Indexes:
      PrimaryKey
        Keys:
          x
          y
        Condition: ((x, y) in 2-element set)
        Parts: 1/1
        Granules: 3/6

11 rows in set. Elapsed: 0.016 sec.
```

### CODE REVIEW

File: src/Storages/MergeTree/KeyCondition.cpp [hunk 1]
```diff
@@ -2329,7 +2341,17 @@ bool KeyCondition::mayBeTrueInRange(
...
+String KeyCondition::RPNElement::toString() const {
+    if (key_column.size() == 1) {
+        return toString("column " + std::to_string(key_column[0]), false);
+    } else if (key_column.size() > 1) {
+        WriteBufferFromOwnString buf;
+        for (size_t i : key_column) {
+            buf << i << " ";
+        }
+    }
+    return "no columns";
```
Comment (mgill25) [inline src/Storages/MergeTree/KeyCondition.cpp:2353]: Maybe this isn't the correct thing to return here.

**Comment by mgill25** [comment]
Testing is still pending. I would like to write a test for this as well.

**Comment by nickitat** [comment]
pls address style check issues: https://s3.amazonaws.com/clickhouse-test-reports/48871/af010b62f866c6abe6880d98bae2c989220b2838/style_check/style_output.txt

we place braces on a new line. you could install clang-format and it will do it for you.

**Comment by alexey-milovidov** [comment]
> `  in 2-element set`

Double whitespace.

**Comment by mgill25** [comment]
Just FYI I haven't forgotten about this. Looks like my changes broke quite a few tests. I'll try to find out what went wrong and fix it. Just need to find some time :)

**Pull request closed without merging**